backend/src/scripts/utility/teamspaces/removeIncompleteRevisions.js

```diff
@@ -0,0 +1,186 @@
+	await processFilesAndRefs(
+		teamspace,
+		`${model}.stash.json_mpc.ref`,
+		undefined,
+		{ _id: { $regex: `^${UUIDToString(revId)}/` } },
+	);
+
+	const unity3d = await find(teamspace, `${model}.stash.unity3d`, { _id: revId }, { assets: 1, jsonFiles: 1 });
+	for (const { assets, jsonFiles } of unity3d) {
+		// eslint-disable-next-line no-await-in-loop
+		await processFilesAndRefs(teamspace, `${model}.stash.unity3d.ref`, assets.map((filename) => filename.replace(`/${teamspace}/${model}/`, '')));
+		// eslint-disable-next-line no-await-in-loop
+		await processFilesAndRefs(teamspace, `${model}.stash.json_mpc.ref`, jsonFiles.map((filename) => filename.replace(`/${teamspace}/${model}/`, '')));
+		// eslint-disable-next-line no-await-in-loop
+		await processFilesAndRefs(teamspace, `${model}.stash.json_mpc.ref`, assets.map((filename) => filename.replace(`/${teamspace}/${model}/`, '').replace('unity3d', 'json.mpc')));
+		// eslint-disable-next-line no-await-in-loop
+		await processFilesAndRefs(teamspace, `${model}.stash.src.ref`, assets.map((filename) => filename.replace(`/${teamspace}/${model}/`, '').replace('unity3d', 'src.mpc')));
+	}
+
+	await processCollection(teamspace, `${model}.stash.unity3d`, { _id: revId }, '_extRef');
+	await processCollection(teamspace, `${model}.stash.3drepo`, { rev_id: revId }, '_extRef');
+};
+
+const processModelSequences = async (teamspace, model, revId) => {
+	const sequences = await find(teamspace, `${model}.sequences`, { rev_id: revId }, { frames: 1 });
+	const sequenceFilenames = [];
+
+	for (const { _id, frames } of sequences) {
+		if (frames) {
+			for (const { state } of Object.values(frames)) {
+				sequenceFilenames.push(state);
+				// eslint-disable-next-line no-await-in-loop
+				await processCollection(teamspace, `${model}.activities`, { sequenceId: _id }, '_extRef');
+				// eslint-disable-next-line no-await-in-loop
+				await processFilesAndRefs(teamspace, `${model}.activities.ref`, undefined, { _id: UUIDToString(_id) });
+			}
+		}
+	}
+
+	await processFilesAndRefs(teamspace, `${model}.sequences.ref`, sequenceFilenames);
+
+	await deleteMany(teamspace, `${model}.sequences`, { rev_id: revId });
+};
+
+const processModelScene = async (teamspace, model, revId) => {
+	await processCollection(teamspace, `${model}.scene`, { rev_id: revId }, '_extRef');
+};
+
+const processTeamspace = async (teamspace) => {
+	const cols = await getCollectionsEndsWith(teamspace, '.history');
+
+	for (const { name } of cols) {
+		const incompleteRevisionFilter = {
+			incomplete: { $exists: true },
+			timestamp: { $lt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 14) },
+		};
+		// eslint-disable-next-line no-await-in-loop
+		const badRevisions = await find(teamspace, name, incompleteRevisionFilter, { rFile: 1 });
+
+		for (const { _id, rFile } of badRevisions) {
+			const model = name.slice(0, -('.history'.length));
+			logger.logInfo(`\t\t-${model}::${UUIDToString(_id)}`);
+
+			// eslint-disable-next-line no-await-in-loop
```

Comment:
Have a think what can be done in parallel, I don't think you need to await for everything like this